cmd/app/link_test.go

```diff
@@ -537,6 +538,159 @@ func Test_Apps_Link(t *testing.T) {
 				require.Len(t, apps, 0)
 			},
 		},
+		"manifest source prompt should not display for Run-on-Slack apps with local manifest source": {
+			Setup: func(t *testing.T, ctx context.Context, cm *shared.ClientsMock, cf *shared.ClientFactory) {
+				cm.AuthInterface.On("Auths", mock.Anything).Return([]types.SlackAuth{
+					mockLinkSlackAuth1,
+					mockLinkSlackAuth2,
+				}, nil)
+				cm.AddDefaultMocks()
+				setupAppLinkCommandMocks(t, ctx, cm, cf)
+				// Set manifest source to local
+				if err := cm.Config.ProjectConfig.SetManifestSource(ctx, config.MANIFEST_SOURCE_LOCAL); err != nil {
+					require.FailNow(t, fmt.Sprintf("Failed to set the manifest source in the memory-based file system: %s", err))
+				}
+				// Mock manifest for Run-on-Slack app
+				manifestMock := &app.ManifestMockObject{}
+				manifestMock.On("GetManifestLocal", mock.Anything, mock.Anything).Return(types.SlackYaml{
+					AppManifest: types.AppManifest{
+						Settings: &types.AppSettings{
+							FunctionRuntime: types.SLACK_HOSTED,
+						},
+					},
+				}, nil)
+				cf.AppClient().Manifest = manifestMock
+				cm.IO.On("SelectPrompt",
+					mock.Anything,
+					"Select the existing app team",
+					mock.Anything,
+					mock.Anything,
+					mock.Anything,
+				).Return(iostreams.SelectPromptResponse{
+					Flag:   true,
+					Option: mockLinkSlackAuth1.TeamDomain,
+				}, nil)
+				cm.IO.On("InputPrompt",
+					mock.Anything,
+					"Enter the existing app ID",
+					mock.Anything,
+				).Return(mockLinkAppID1, nil)
+				cm.IO.On("SelectPrompt",
+					mock.Anything,
+					"Choose the app environment",
+					mock.Anything,
+					mock.Anything,
+					mock.Anything,
+				).Return(iostreams.SelectPromptResponse{
+					Flag:   true,
+					Option: "deployed",
+				}, nil)
+				cm.ApiInterface.On(
+					"GetAppStatus",
+					mock.Anything,
+					mockLinkSlackAuth1.Token,
+					[]string{mockLinkAppID1},
+					mockLinkSlackAuth1.TeamID,
+				).Return(api.GetAppStatusResult{}, nil)
+			},
+			CmdArgs: []string{},
+			ExpectedAsserts: func(t *testing.T, ctx context.Context, cm *shared.ClientsMock) {
+				expectedApp := types.App{
+					AppID:        mockLinkAppID1,
+					TeamDomain:   mockLinkSlackAuth1.TeamDomain,
+					TeamID:       mockLinkSlackAuth1.TeamID,
+					EnterpriseID: mockLinkSlackAuth1.EnterpriseID,
+				}
+				actualApp, err := cm.AppClient.GetDeployed(
+					ctx,
+					mockLinkSlackAuth1.TeamID,
+				)
+				require.NoError(t, err)
+				assert.Equal(t, expectedApp, actualApp)
+				// Assert manifest confirmation prompt was not displayed
+				cm.IO.AssertNotCalled(t, "ConfirmPrompt",
+					mock.Anything,
+					LinkAppManifestSourceConfirmPromptText,
+					mock.Anything,
+				)
+			},
+		},
+		"manifest source prompt should display for GBP apps with local manifest source": {
```

Comment:
note: Added a test-case to assert that the manifest update prompt is still displayed for GBP apps with a local manifest. This case might be redundant, but I felt high confidence after adding it.